src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -834,6 +869,18 @@ private bool InferConnectionDetails()
                     ReportSNIError(SNIProviders.INVALID_PROV);
                     return false;
                 }
+
+                if (ConnectionProtocol == DataSource.Protocol.None)
```

Comment:
I will fix it as you mentioned for now, but I strongly believe we should discuss about it, and arrive to one agreement.